modules/fedimint-ln-client/src/lib.rs

```diff
@@ -910,7 +916,20 @@ impl LightningClientModule {
     ///
     /// See also [`Self::update_gateway_cache_continuously`].
     pub async fn update_gateway_cache(&self) -> anyhow::Result<()> {
-        let gateways = self.module_api.fetch_gateways().await?;
+        let Ok(mut guard) = self.gateway_cache_update_mutex.try_lock() else {
```

Comment:
I am not sure if this is correct, and it rings a spider-sense, that it might be unnecessarily non-trivial.

As it is RN, e.g. I don't see any `.take()` or error being overwritten with `None` so where is the error ever cleared? 

The chances that this has contention seems rather small, so why not just lock , and not merge?

Reply:
Alternatively, if you really, relaly want to merge, then I would do it with an extra `AtomicU64` counter that would get incremented on every success. `load` counter before locking, after locking check if it increased, if it did return `Ok(())`, otherwise let the function try again, as the previous one must have failed. Avoids `e.to_string()` shenigenans which I in particuarlly don't like as they potentially break error chains, etc. which might lead to inconsitent, non-deterministric operation.

Reply:
Yeah I missed this last time I looked. IMO the gateway cache is just best effort, we shouldn't try to exactly synchronize with what is registered with the federation. I'm in favoring of just locking and avoiding merging.